modules/dailymotionBidAdapter.js

```diff
@@ -130,76 +130,96 @@ export const spec = {
-          atts: deepAccess(bidderRequest, 'ortb2.device.ext.atts', 0),
-        },
-      } : {}),
-      request: {
-        adUnitCode: deepAccess(bid, 'adUnitCode', ''),
-        auctionId: deepAccess(bid, 'auctionId', ''),
-        bidId: deepAccess(bid, 'bidId', ''),
-        mediaTypes: {
-          video: {
-            api: bid.mediaTypes?.[VIDEO]?.api || [],
-            mimes: bid.mediaTypes?.[VIDEO]?.mimes || [],
-            minduration: bid.mediaTypes?.[VIDEO]?.minduration || 0,
-            maxduration: bid.mediaTypes?.[VIDEO]?.maxduration || 0,
-            playbackmethod: bid.mediaTypes?.[VIDEO]?.playbackmethod || [],
-            plcmt: bid.mediaTypes?.[VIDEO]?.plcmt || 1, // Fallback to instream considering logic of `isBidRequestValid`
-            protocols: bid.mediaTypes?.[VIDEO]?.protocols || [],
-            skip: bid.mediaTypes?.[VIDEO]?.skip || 0,
-            skipafter: bid.mediaTypes?.[VIDEO]?.skipafter || 0,
-            skipmin: bid.mediaTypes?.[VIDEO]?.skipmin || 0,
-            startdelay: bid.mediaTypes?.[VIDEO]?.startdelay || 0,
-            w: bid.mediaTypes?.[VIDEO]?.w || 0,
-            h: bid.mediaTypes?.[VIDEO]?.h || 0,
+        // Cast boolean in any case (value should be 0 or 1) to ensure type
+        coppa: !!deepAccess(bidderRequest, 'ortb2.regs.coppa'),
+        // In app context, we need to retrieve additional informations
+        ...(!deepAccess(bidderRequest, 'ortb2.site') && !!deepAccess(bidderRequest, 'ortb2.app') ? {
+          appBundle: deepAccess(bidderRequest, 'ortb2.app.bundle', ''),
+          appStoreUrl: deepAccess(bidderRequest, 'ortb2.app.storeurl', ''),
+        } : {}),
+        ...(deepAccess(bidderRequest, 'ortb2.device') ? {
+          device: {
+            lmt: deepAccess(bidderRequest, 'ortb2.device.lmt', null),
+            ifa: deepAccess(bidderRequest, 'ortb2.device.ifa', ''),
+            atts: deepAccess(bidderRequest, 'ortb2.device.ext.atts', 0),
+          },
+        } : {}),
+        request: {
+          adUnitCode: deepAccess(bid, 'adUnitCode', ''),
+          auctionId: deepAccess(bid, 'auctionId', ''),
+          bidId: deepAccess(bid, 'bidId', ''),
+          mediaTypes: {
+            video: {
+              api: bid.mediaTypes?.[VIDEO]?.api || [],
+              mimes: bid.mediaTypes?.[VIDEO]?.mimes || [],
+              minduration: bid.mediaTypes?.[VIDEO]?.minduration || 0,
+              maxduration: bid.mediaTypes?.[VIDEO]?.maxduration || 0,
+              playbackmethod: bid.mediaTypes?.[VIDEO]?.playbackmethod || [],
+              plcmt: bid.mediaTypes?.[VIDEO]?.plcmt || 1, // Fallback to instream considering logic of `isBidRequestValid`
+              protocols: bid.mediaTypes?.[VIDEO]?.protocols || [],
+              skip: bid.mediaTypes?.[VIDEO]?.skip || 0,
+              skipafter: bid.mediaTypes?.[VIDEO]?.skipafter || 0,
+              skipmin: bid.mediaTypes?.[VIDEO]?.skipmin || 0,
+              startdelay: bid.mediaTypes?.[VIDEO]?.startdelay || 0,
```

Comment:
It is not appropriate to fill this in with a fallback; preroll must be declared by the publisher. You're putting their inventory at risk for getting banned by DSPs for incorrect declarations.